### ISSUE TITLE
GitHub releases and/or tags on non-master branch

### ISSUE THREAD

**Issue by jtwee** [YahnisElsts/plugin-update-checker#290]
I noticed in the GitHubApi class the [chooseReference](https://github.com/YahnisElsts/plugin-update-checker/blob/3f5a340ded47653949cc14c3a821027d9fac5416/Puc/v4p6/Vcs/GitHubApi.php#L314) function only checks for releases or tags if the branch is set to `master`. Is there any harm in checking for them on other branches?

My use case was for having releases occur on a separate branch called `updates` that can filter out some of the file noise from `master`.

I'd be happy to do a PR for it but wanted to check first whether this fits with your vision.

**Comment by YahnisElsts** [comment]
This update checker was designed with the idea that it should require as little configuration as possible. If you're using releases or tags, you don't need to do anything special because that's already the default assumption. If you're treating `master` as the stable branch, you also don't need to do anything because that's the automatic fallback. You only need to set a specific branch if you're *not* using releases or tags and you have some kind of a `release` branch instead.

In other words:
- If you *are* using releases or tags, there's no need to give the update checker a branch name. 
- if you *are not* using releases or tags and you use an `updates` branch instead, there's no point in checking for releases and tags.

**Comment by jtwee** [comment]
Thanks for the clarification, I didn't quite get that from the documentation.
